src/transport/message_handler.rs

```diff
@@ -0,0 +1,210 @@
+use crate::events::{Event, EventHandler};
+use crate::transport::msgs::{LSPSMessage, Prefix, RawLSPSMessage, LSPS_MESSAGE_TYPE};
+use bitcoin::secp256k1::PublicKey;
+use lightning::ln::peer_handler::CustomMessageHandler;
+use lightning::ln::wire::CustomMessageReader;
+use lightning::log_info;
+use lightning::util::logger::Logger;
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::io;
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+/// A trait used to implement a specific LSPS protocol
+/// The messages the protocol uses need to be able to be mapped
+/// from and into LSPSMessages.
+pub trait ProtocolMessageHandler {
+	type ProtocolMessage: TryFrom<LSPSMessage> + Into<LSPSMessage>;
+	const PROTOCOL_NUMBER: Option<u16>;
+
+	fn handle_message(
+		&self, message: Self::ProtocolMessage, counterparty_node_id: &PublicKey,
+	) -> Result<(), lightning::ln::msgs::LightningError>;
+	fn get_and_clear_pending_protocol_messages(&self) -> Vec<(PublicKey, Self::ProtocolMessage)>;
+	fn get_and_clear_pending_protocol_events(&self) -> Vec<Event>;
+	fn get_protocol_number(&self) -> Option<u16> {
+		Self::PROTOCOL_NUMBER
+	}
+}
+
+/// A trait used to implement the mapping from a LSPS transport layer mesage
+/// to a specific protocol message. This enables the ProtocolMessageHandler's
+/// to not need to know about LSPSMessage and only have to deal with the specific
+/// messages related to the protocol that is being implemented.
+pub trait TransportMessageHandler {
```

Comment:
I'm a bit dubious why we need this second trait exactly? Since it's not used in the LSPS2 PR, is it just there for the blanket impl? Couldn't the blanket impl be added to `ProtocolMessageHandler`? 

If we need it somehow, should we be adding the type constraint here?
```suggestion
pub trait TransportMessageHandler: ProtocolMessageHandler {
```